main.py

```diff
@@ -326,11 +325,23 @@ def _add_flow_store(self, flow_dict, switch):
         if switch.id not in stored_flows_box:
             stored_flows_box[switch.id] = OrderedDict()
 
-        # TODO handle issue 23 (overlapping FlowMod add)
         if not stored_flows_box[switch.id].get(cookie):
             stored_flows_box[switch.id][cookie] = [installed_flow]
         else:
-            stored_flows_box[switch.id][cookie].append(installed_flow)
+            version = switch.connection.protocol.version
+            stored_flows = stored_flows_box[switch.id].get(cookie, [])
+            for i, _ in enumerate(stored_flows):
+                if all(
+                    (
+                        stored_flows[i]["flow"].get("priority", 0)
+                        == flow_dict.get("priority", 0),
+                        match_flow(flow_dict, version, stored_flows[i]["flow"]),
+                    )
+                ):
+                    stored_flows_box[switch.id][cookie][i] = installed_flow
+                    break
+            else:
+                stored_flows_box[switch.id][cookie].append(installed_flow)
```

Comment:
Why `enumerate` and then using just the index?

Reply:
Good catch. I meant to also use the `v` value (`stored_flow`) this way here https://github.com/kytos-ng/flow_manager/pull/44/commits/5b667e9290a10ed4c95f2e991570a84f99defb12, I've just updated this.